homeassistant/components/vesync/light.py

```diff
@@ -68,17 +88,79 @@ def turn_on(self, **kwargs):
             brightness = round((brightness / 255) * 100)
             # clamp to 1-100
             brightness = max(1, min(brightness, 100))
-            self.dimmer.set_brightness(brightness)
+            self.device.set_brightness(brightness)
         # Avoid turning device back on if this is just a brightness adjustment
         if not self.is_on:
             self.device.turn_on()
 
     @property
-    def supported_features(self):
-        """Get supported features for this entity."""
-        return SUPPORT_BRIGHTNESS
+    def color_mode(self):
+        """Set color mode for this entity."""
+        return COLOR_MODE_BRIGHTNESS
 
     @property
-    def brightness(self):
-        """Get dimmer brightness."""
-        return round((int(self.dimmer.brightness) / 100) * 255)
+    def supported_color_modes(self):
+        """Flag supported color_modes (in an array format)."""
+        return [COLOR_MODE_BRIGHTNESS]
+
+
+class VeSyncTunableWhiteLightHA(VeSyncBaseLight, LightEntity):
+    """Representation of a VeSync Tunable White Light device."""
+
+    def __init__(self, device):
+        """Initialize the VeSync Tunable White Light device."""
+        super().__init__(device)
+        self.device = device
+
+    def turn_on(self, **kwargs):
+        """Turn the device on."""
+        if ATTR_BRIGHTNESS in kwargs:
+            # get brightness from HA data
+            brightness = int(kwargs[ATTR_BRIGHTNESS])
+            # convert to percent that vesync api expects
+            brightness = round((brightness / 255) * 100)
+            # clamp to 1-100
+            brightness = max(1, min(brightness, 100))
+            # pass value to pyvesync library api
+            self.device.set_brightness(brightness)
+        if ATTR_COLOR_TEMP in kwargs:
+            # get brightness from HA data
+            color_temp = int(kwargs[ATTR_COLOR_TEMP])
+            # flip cold/warm to what pyvesync api expects
+            color_temp = 100 - color_temp
+            # ensure value between 1-100
+            color_temp = max(1, min(color_temp, 100))
+            # pass value to pyvesync library api
+            self.device.set_color_temp(color_temp)
+        # Avoid turning device back on if this is just a brightness adjustment
+        if not self.is_on:
+            self.device.turn_on()
+
+    @property
+    def color_temp(self):
```

Comment:
Color temp needs to be in mireds

https://developers.home-assistant.io/docs/core/entity/light/#properties